clusters/resource_cluster.go

```diff
@@ -472,6 +472,8 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, c *commo
 
 	clusterInfo, err := clusterWaiter.GetWithTimeout(timeout)
 	if err != nil {
+		// In case of "ERROR" or "TERMINATED" state, WaitGetClusterRunning returns an error and we should delete the cluster before returning
+		resourceClusterDelete(ctx, d, c)
```

Comment:
we need to show the error if we can't delete the resource, etc.

Reply:
Like this: https://github.com/databricks/terraform-provider-databricks/blob/main/vectorsearch/resource_vector_search_endpoint.go#L53

Reply:
Yes, I was just testing a few things with directly calling deletion method before sending it in review, will update it.